code/game/objects/structures/mannequin.dm

```diff
@@ -0,0 +1,1187 @@
+				else if(clothToUpdate.icon_override)
+					I = image(clothToUpdate.icon_override, t_state)
+				else
+					I = image(slotIcon[MANNEQUIN_ICONS_SLOT], t_state)
+
+		if(species.name in clothToUpdate.species_fit)
+			var/icon/species_icon = slotIcon[MANNEQUIN_ICONS_SPECIES]
+			if(species_icon)
+				I.icon = species_icon
+
+		if(clothToUpdate.icon_override)
+			I.icon	= clothToUpdate.icon_override
+
+		O.overlays += I
+
+		if(clothToUpdate.dynamic_overlay)
+			if(clothToUpdate.dynamic_overlay["[slotIcon[MANNEQUIN_DYNAMIC_LAYER]]"])
+				var/image/dyn_overlay = clothToUpdate.dynamic_overlay["[slotIcon[MANNEQUIN_DYNAMIC_LAYER]]"]
+				O.overlays += dyn_overlay
+
+		if(clothToUpdate.blood_DNA && clothToUpdate.blood_DNA.len)
+			var/bloodsies_state = get_bloodsies_state(clothToUpdate,slot)
+			if(bloodsies_state)
+				var/image/bloodsies	= image('icons/effects/blood.dmi', bloodsies_state)
+				bloodsies.color		= clothToUpdate.blood_color
+				O.overlays += bloodsies
+
+		clothToUpdate.generate_accessory_overlays(O)
+
+/obj/structure/mannequin/proc/update_icon_hand(var/obj/Overlays/O,var/index)
+	var/obj/item/heldItem = get_held_item_by_index(index)
+
+	if(heldItem)
+		var/t_state = heldItem.item_state
+		var/t_inhand_state = heldItem.inhand_states[get_direction_by_index(index)]
+		var/icon/check_dimensions = new(t_inhand_state)
+		if(!t_state)
+			t_state = heldItem.icon_state
+
+		var/image/I  = image(t_inhand_state, t_state)
+		I.pixel_x = -1*(check_dimensions.Width() - WORLD_ICON_SIZE)/2
+		I.pixel_y = -1*(check_dimensions.Height() - WORLD_ICON_SIZE)/2
+
+		var/list/offsets = get_item_offset_by_index(index)
+
+		I.pixel_x += offsets["x"]
+		I.pixel_y += offsets["y"]
+
+		if(heldItem.dynamic_overlay && heldItem.dynamic_overlay["[HAND_LAYER]-[index]"])
+			var/image/dyn_overlay = heldItem.dynamic_overlay["[HAND_LAYER]-[index]"]
+			O.overlays += dyn_overlay
+
+		O.overlays += I
+
+/obj/structure/mannequin/proc/get_all_slot_icons()
+	all_slot_icons[SLOT_MANNEQUIN_ICLOTHING] = get_slot_icons(SLOT_MANNEQUIN_ICLOTHING)
+	all_slot_icons[SLOT_MANNEQUIN_FEET] = get_slot_icons(SLOT_MANNEQUIN_FEET)
+	all_slot_icons[SLOT_MANNEQUIN_GLOVES] = get_slot_icons(SLOT_MANNEQUIN_GLOVES)
+	all_slot_icons[SLOT_MANNEQUIN_EARS] = get_slot_icons(SLOT_MANNEQUIN_EARS)
+	all_slot_icons[SLOT_MANNEQUIN_OCLOTHING] = get_slot_icons(SLOT_MANNEQUIN_OCLOTHING)
+	all_slot_icons[SLOT_MANNEQUIN_EYES] = get_slot_icons(SLOT_MANNEQUIN_EYES)
+	all_slot_icons[SLOT_MANNEQUIN_BELT] = get_slot_icons(SLOT_MANNEQUIN_BELT)
+	all_slot_icons[SLOT_MANNEQUIN_MASK] = get_slot_icons(SLOT_MANNEQUIN_MASK)
+	all_slot_icons[SLOT_MANNEQUIN_HEAD] = get_slot_icons(SLOT_MANNEQUIN_HEAD)
+	all_slot_icons[SLOT_MANNEQUIN_BACK] = get_slot_icons(SLOT_MANNEQUIN_BACK)
+	all_slot_icons[SLOT_MANNEQUIN_ID] = get_slot_icons(SLOT_MANNEQUIN_ID)
+
+/obj/structure/mannequin/proc/get_slot_icons(var/slot)
+	var/list/slotIcon = list()
+	switch(slot)
+		if(SLOT_MANNEQUIN_ICLOTHING)
```

Comment:
I'll be waiting for someone to rework mob inventories before investing time in designing polymorphic procs that only mannequins would use anyway. I'll put this PR on hold if this is what it takes.